python/ray/experimental/sgd/mnist_example.py

```diff
@@ -101,14 +112,16 @@ def train_mnist(config, reporter):
         strategy=args.strategy)
 
     # Important: synchronize the initial weights of all model replicas
-    w0 = sgd.for_model(lambda m: m.variables.get_flat())
-    sgd.foreach_model(lambda m: m.variables.set_flat(w0))
+    w0 = sgd.for_model(lambda m: m.get_variables().get_flat())
+    sgd.foreach_model(lambda m: m.get_variables().set_flat(w0))
 
     for i in range(args.num_iters):
         if i % 10 == 0:
             start = time.time()
             loss = sgd.step(fetch_stats=True)["loss"]
-            acc = sgd.foreach_model(lambda model: model.test_accuracy())
+            metrics = sgd.foreach_model(lambda model: model.get_metrics())
+            #TODO metrics worker_num * device_num matrix
```

Comment:
@ericl It seems a little strange here. `foreach_model` returns a matrix which  size equals to worker_size * device_size. Previously in `DistributedSGD::foreach_model`,  extend is used to compact them into a vector of worker_size.  How to handle the case when the function passed to foreach_model returns dict ?

Reply:
Got it. Have updated the code.